From 3f84d868d95cd04c753351f7fd8536869a919e0b Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Tue, 8 Jul 2014 23:04:30 +0300
Subject: Downgrade the optimization of regcomp.c on mips due to a gcc-4.9 bug

  % cat ~/t.pl
  print ( (q{foo'} =~ /[^']+'/) ?  "ok\n" : "not ok\n" );
  % perl ~/t.pl
  ok
  % ./miniperl ~/t.pl
  not ok

This only happens with -fPIC -ftree-vrp builds.
It is a regression from gcc-4.8.

Bug-Debian: http://bugs.debian.org/754054
Patch-Name: debian/regcomp-mips-optim.diff
---
 cflags.SH | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cflags.SH b/cflags.SH
index 5363f99..4f2794f 100755
--- a/cflags.SH
+++ b/cflags.SH
@@ -275,6 +275,11 @@ for file do
     : or customize here
 
     case "$file" in
+    regcomp) : work around http://bugs.debian.org/754054
+        case $archname in
+        mips-*)
+            optimize="$optimize -fno-tree-vrp";;
+        esac;;
     *) ;;
 
     *) : Customization examples follow: ;;
